### ISSUE TITLE
input messes up if ctrl-c a atuin run command - have to restart ssh/terminal session to fix (bash )

### ISSUE THREAD

**Issue by bob454522** [atuinsh/atuin#996]
(using atuin 14.0.1  built from main in past week - ie ~ may 13 2023 main build)

I can re-produce this on various VMs / OSes im using with atuin (namely - debian8 and Ubuntu 18 and ubuntu20  , all running bash)-

if you get your command from atuin,  press enter (to execute that command),  **then interrupt command while its running (ctrl-c)** ,  you are left with an unusual-able terminal/ssh session,  and have to exit and reconnect.  (im guessing echo of typed commands gets messed up or disabled some how).  I have also seen this occur, on its own,  when an atuin provided command either fails on its own or similar (like a bash script that exits prematurely,  not every time though)

Ive almost never encountered this issue before (so im not sure how to describe / troubleshoot it).  to reproduce this:

`sleep 10s; echo "sleep done"`

(while its sleeping 10sec, hit ctrl-c )  -  should be no problem.
now access atuin (up-arrow,  ctrl-r) ,  and have atuin dump in that same sleep 10s; echo   command,  and while its running ,  hit ctrl-c (to interuput the 10sec of sleep).     now my termanl is now echoing back,  and i have to exit and reconnect.

video showing this (animated gif below).  (im using mobaXterm mainly and have this issue,  but i can reproduce this with putty as well)
(note towards the end im typing but you cant see it,  i do type `echo hi`  but you only see `hi`
(os seen in the animated gif below is Ubuntu 18.04 LTS)
(LOVE AUTIN btw!!)

![atuin term gif](https://github.com/ellie/atuin/assets/60830628/975a96c8-da06-4810-bf61-1462d6b41547)

**Comment by PMeira** [comment]
Just noticed that other interruptions like segfaults in the process also break the shell output, not only Ctrl+C.

**Comment by itszn** [comment]
I have a workaround that seems to alleviates the issue. Having the bash precmd run `stty echo` and `stty sane` seems to return the cursor back to its expected function.


I still need to play with it some more to make sure it doesn't have any other side effects.

Update your bashrc to include the code after your `atuin init`
```bash
eval "$(atuin init bash --disable-up-arrow)"

# Add the code below

_atuin_fix_echo() {
    stty echo
    stty sane
}

if [[ -n "${BLE_VERSION-}" ]]; then
    blehook PRECMD-+=_atuin_fix_echo
else
    precmd_functions+=(_atuin_fix_echo)
fi
```
cc @PMeira does this also fix the issue for the other cases you noticed?

**Comment by PMeira** [comment]
@itszn It does fix it for the situations I found so far. Thanks!

**Comment by darthShadow** [comment]
Seems like a duplicate of #492

**Comment by PMeira** [comment]
@darthShadow Indeed! I installed bash 5.2.15 to confirm and it seems to be fixed without workarounds, at least for the basic Ctrl+C test.

It's likely that many distros don't ship with bash 5.2 yet, so maybe it's worth linking that issue in https://github.com/ellie/atuin#bash to save users some time.

**Comment by Amar1729** [comment]
FWIW: as a temporary workaround, the `reset` command seems to fix the shell after input becomes broken (although the characters don't actually show up, you can still type "reset<enter>"). Avoids you having to close/re-open a new shell.

**Comment by ellie** [comment]
Closed as a dupe of #492
